### PR TITLE
Support @room pills while editing

### PR DESCRIPTION
**Pull request by bwindels**
https://github.com/vector-im/riot-web/issues/9840

atroom

### CODE REVIEW

File: src/editor/deserialize.js [hunk 1]
```diff
@@ -16,73 +16,83 @@ limitations under the License.
...
 
-function parseLink(a, room, client) {
+function parseAtRoomMentions(text, partCreator) {
+    const ATROOM = "@room";
+    const parts = [];
+    text.split(ATROOM).forEach((textPart, i, arr) => {
+        if (textPart.length) {
+            parts.push(partCreator.plain(textPart));
+        }
+        const isLast = i === arr.length - 1;
```
Comment (turt2live) [inline src/editor/deserialize.js]: This variable feels a bit misleading, as I was about to comment that it seems to filter out `@room` pills in the middle of the text. However, some black magic causes them to work just fine... somehow?

A comment to explain might also help.

ftr I was testing with this text: `"@room check out the @room changes I made to the @room"`

File: src/editor/parts.js [hunk 2]
```diff
@@ -270,7 +260,14 @@ export class RoomPillPart extends PillPart {
...
 }
 
-export class UserPillPart extends PillPart {
+class AtRoomPillPart extends RoomPillPart {
+    get type() {
+        return "at-room-pill";
+    }
+}
+
+
```
Comment (turt2live) [inline src/editor/parts.js]: extreme nit: too many blank lines

File: src/editor/deserialize.js (hunk 1 above)
Comment (bwindels) [reply src/editor/deserialize.js]: We split the string with `"@room"` as a separator, so in the above case, we'd get the following array back:
```
[ '', ' check out the ', ' changes I made to the ', '' ]
```
E.g. an empty string before the first, and after the last encounter of the separator. So no need to append a pill for the last match.

Wonder how to best make this clear ... I added a comment, but feel free to suggest any naming if you'd prefer.

Comment (turt2live) [reply src/editor/deserialize.js]: ah ha, the comment does help. Thank you!

**Pull request merged**